arangod/GeneralServer/HttpCommTask.cpp

```diff
@@ -795,9 +795,9 @@ ResponseCode HttpCommTask::handleAuthHeader(HttpRequest* req) const {
 
       req->setAuthenticationMethod(authMethod);
       if (authMethod != AuthenticationMethod::NONE) {
-        auto entry = _auth->tokenCache().checkAuthentication(authMethod, auth);
-        req->setAuthenticated(entry.authenticated());
-        req->setUser(std::move(entry._username));
+        _authToken = _auth->tokenCache().checkAuthentication(authMethod, auth);
+        req->setAuthenticated(_authToken.authenticated());
+        req->setUser(std::move(_authToken._username));
```

Comment:
If we keep the authToken as an instance variable, should we still move the username out of it?

Reply:
We keep it as a instance variable of `HttpCommTask` and here it is copied into the `GeneralRequest` struct. Those are somewhat independent.

Reply:
I don't this.
However, it is unexpected to have an instance variable that has moved-out members. So if anyone later uses the instance variable, it will be in an incomplete state. I am therefore suggesting to not move the data out, but to copy it out.